File: src/CashFlow.Host/ClientApp/src/models/transaction.ts
import { FinancialYear } from './financial-year';
import { Supplier } from './supplier';

export class TransactionCode {
  codeName!: string;
  transactionId!: string;
  dateAssigned!: Date;
}

export class Transaction {
  id!: string;
  evidenceNumber!: number;
  accountId!: string;
  transactionDate: Date;
  amountInCents!: number;
  isInternalTransfer!: boolean;
  description!: string;
  comment: string;
  codes!: TransactionCode[];
  financialYear: FinancialYear;
  supplier: Supplier;
}

File: src/CashFlow.Data.Abstractions/Entities/Transaction.cs
using System;
using System.Collections.Generic;

namespace CashFlow.Data.Abstractions.Entities
{
    public sealed class Transaction
    {
        public Guid Id { get; set; }

        public DateTimeOffset TransactionDate { get; set; }

        public string EvidenceNumber { get; set; }

        public Guid FinancialYearId { get; set; }

        public Guid AccountId { get; set; }

        public Guid? SupplierId { get; set; }

        public DateTimeOffset DateCreated { get; set; }

        public DateTimeOffset? DateModified { get; set; }

        public long AmountInCents { get; set; }

        public bool IsInternalTransfer { get; set; }

        public string Description { get; set; }

        public string Comment { get; set; }

        public ICollection<TransactionCode> Codes { get; set; }
    }
}
